### DISCUSSION TITLE
MUC/Sub does not deliver notifications to the user when offline

### DISCUSSION THREAD

**Discussion by karimhm** [processone/ejabberd#3663] [Questions] [answered]
When I subscribe a user to a room ejabberd does not send push notification to the user.

I create the room with the following REST API call

**`POST`** `{{host}}/api/create_room_with_opts`
```json
{
    "name": "room_1",
    "service": "conference.****.com",
    "host": "****.com",
    "options": {
        "allow_subscription": "true"
    }
}
```
I receive HTTP 200 (OK) status.

Then I subscribe a user with the following REST API call
```json
{
    "user": "100770926037933908458@.****.com",
    "nick": "5850FA9D-B961-4FB1-8A42-7B1DA5EE0C8B",
    "room": "room_1@conference.****.com",
    "nodes": "urn:xmpp:mucsub:nodes:messages"
}
```

I receive 
```json
[
    "urn:xmpp:mucsub:nodes:messages"
]
```

But when a new message is sent to the `room_1` MUC room a push notification is not triggered. Push notification are triggered for standard 1:1 messages or when the user enters an MUC room in the sense of [XEP-0045](https://xmpp.org/extensions/xep-0045.html) .

**Comment by weiss** [comment]
Which ejabberd version?

**Reply by karimhm** [reply]
**21.04**

**Comment by karimhm** [comment, marked as answer]
As pointed by @weiss at #3662 this should be fixed by 10905b0447f3339a9b6fd4b722d6ccba2fe1b7f7 in ejabberd **21.07**.
